rabbitmq_amqp_python_client/connection.py

```diff
@@ -80,17 +100,47 @@ def dial(self) -> None:
             logger.debug("Enabling SSL")
 
             self._ssl_domain = SSLDomain(SSLDomain.MODE_CLIENT)
-            if self._ssl_domain is not None:
-                self._ssl_domain.set_trusted_ca_db(self._conf_ssl_context.ca_cert)
+            assert self._ssl_domain
```

Comment:
I don't know why mypy needs this assert in order to assume it can't be `None` given that it is initialized to an instance of `SSLDomain` above, or never reaching this line due to the constructor raising an exception. But is does :(